### DISCUSSION TITLE
Hardware Compatability

### DISCUSSION THREAD

**Discussion by jramborger78** [talmolab/sleap#1532] [General]
Hey guys,

Would the latest Macbook Air with M2 chip be a worthy item to use SLEAP on? Specifically, we have workstations I can remotely access so the M2 should assist there as well, but my thinking was given its CPU and GPU it could handle some decent processing. Thoughts?

Also, I see that there is a breakdown on analyses using Python, is there one for MATLAB by chance? Or would you suggest more so the Python route? And possibly a sillier question is what enviorment should that be used it? after activating sleap? Conda? OS's terminal? Thanks!

**Comment by roomrys** [comment]
Hi @jramborger78,

I can speak on the M2 front. In lab, we have a Mac Mini 8G 256G to use for troubleshooting datasets on the mac front. We haven't used this for any super large projects (but will need to test its boundaries soon...), but it has at least held up during troubleshooting. I personally invested in an M2 laptop recently because I wanted to be able to develop on SLEAP somewhere other than a desk sometimes....

We don't have explicit instruction for analysis in MATLAB, but the h5 file can definitely be read in with MATLAB and the process should be relatively parallel to the instruction in python (we even use `matplotlib` for the plots). If you are more comfortable/prefer MATLAB, then I would use MATLAB. Feel free to ask questions if you run into trouble!

Thanks,
Liezl

**Comment by jramborger78** [comment]
Awesome, thank you so much. Yeah was basically looking into maybe a M2 with
like 16GB of RAM to be able to smoothly remotely connect to a workstation
on AnyDesk, as well as run inferences from a model, maybe even train one
with 120 videos to then run inferences on subsequent videos. I figured its
cpu and gpu capabilities would be acceptable as my little 8GB Intel i5 was
able to handle 16 videos and 1000 labels acceptably but nothing too
demanding.

I started to teach myself MATLAB due to the need to transpose but after
uploading an h5 it was not what I expected and saw that online you have a
python description.

On Thu, Oct 5, 2023 at 9:33 AM Liezl Maree ***@***.***> wrote:

> Hi @jramborger78 <https://github.com/jramborger78>,
>
> I can speak on the M2 front. In lab, we have a Mac Mini 8G 256G to use for
> troubleshooting datasets on the mac front. We haven't used this for any
> super large projects (but will need to test its boundaries soon...), but it
> has at least held up during troubleshooting. I personally invested in an M2
> laptop recently because I wanted to be able to develop on SLEAP somewhere
> other than a desk sometimes....
>
> We don't have explicit instruction for analysis in MATLAB, but the h5 file
> can definitely be read in with MATLAB and the process should be relatively
> parallel to the instruction in python (we even use matplotlib for the
> plots). If you are more comfortable/prefer MATLAB, then I would use MATLAB.
> Feel free to ask questions if you run into trouble!
>
> Thanks,
> Liezl
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/talmolab/sleap/discussions/1532#discussioncomment-7200784>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BABV7SJ6CHGXFI2RPHFB5BLX53OPFAVCNFSM6AAAAAA5TJXJTSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TEMBQG44DI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Sincerely,

Jarryd Ramborger
951-541-3922
